src/hpp/rostools.py

```diff
@@ -17,8 +18,12 @@ def process_xacro(*args):
 
     opts, input_file_name = xacro.process_args(args)
     try:
+        # If AMENT_PREFIX_PATH is defined (ROS 2), pass the content to retrieve resources
+        ament_prefix_path = os.getenv("AMENT_PREFIX_PATH")
+        if ament_prefix_path is not None:
+            dirs = list(map(lambda s: s + "/share", ament_prefix_path.split(":")))
```

Comment:
Some details:

```suggestion
            dirs = list(map(lambda s: s + "/share", ament_prefix_path.split(":")))
            dirs = [Path(p) / "share" for p in ament_prefix_path.split(os.pathsep)]
```

list comprehensions are more idiomatic than functional paradigm in python. 
Also, os.pathsep is a bit cleaner than hardcoding `:`.
`Path` would be great, but I'm not sure this is allowed by other parts of the API yet. If this is not the case, we can cast those Path objects to `str()`